frontend/tests/features/wheresMyGene.test.ts

```diff
@@ -142,6 +143,69 @@ describe("Where's My Gene", () => {
     }
   });
 
+  test("Source Data", async ({ page }) => {
+    await goToPage(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`, page);
+
+    async function getSourceDataButton() {
+      return page.$(getTestID(SOURCE_DATA_BUTTON_ID));
+    }
+    async function getTissueSelectorButton() {
+      return page.$(getTestID(ADD_TISSUE_ID));
+    }
+
+    async function getGeneSelectorButton() {
+      return page.$(getTestID(ADD_GENE_ID));
+    }
+
+    await clickUntilOptionsShowUp(getTissueSelectorButton, page);
+    await selectFirstOption(page);
+
+    await clickUntilOptionsShowUp(getGeneSelectorButton, page);
+    await selectFirstOption(page);
+
+    await tryUntil(
+      async () => {
+        const canvases = await page.$$("canvas");
+        expect(canvases.length).not.toBe(0);
+      },
+      { page }
+    );
+    await clickUntilSidebarShowsUp(getSourceDataButton, page);
+    await expect(page).toHaveSelector(getText("After filtering cells with low coverage (less than 500 genes expressed)"));
+    const sourceDataList = await page.$("[class*=MuiList-root]");
+    if (!sourceDataList) throw Error("no source data displayed");
+    
+    const sourceDataListItems = await sourceDataList?.$$(".MuiListItem-root")
+    expect(sourceDataListItems?.length).toBe(17);
+
+    await page.mouse.click(0, 0)
+
+    async function getFiltersPanel() {
+      return page.$(getTestID("filters-panel"));
+    }
+    async function getDatasetSelector() {
+      const filtersPanel = await getFiltersPanel();
+
+      if (!filtersPanel) {
+        throw Error("Filters panel not found");
+      }
+
+      return filtersPanel.$("*css=button >> text=Dataset");
+    }    
+    const datasetSelector = await getDatasetSelector();
+    if (!datasetSelector) throw Error("No datasetSelector found");
+    const selectedDatasetsBefore = await datasetSelector.$$(".MuiChip-root");
+    await expect(selectedDatasetsBefore.length).toBe(0);
+    await clickUntilOptionsShowUp(getDatasetSelector, page);
+    await selectFirstOption(page);
+    await clickUntilSidebarShowsUp(getSourceDataButton, page);
+    const sourceDataListAfter = await page.$("[class*=MuiList-root]");
+    if (!sourceDataListAfter) throw Error("no source data displayed after selecting dataset filter");
+    
+    const sourceDataListAfterItems = await sourceDataListAfter?.$$(".MuiListItem-root")
+    expect(sourceDataListAfterItems?.length).toBe(2);   
```

Comment:
Same for this. Maybe just assert > 0?

Reply:
I think this one will always be reliably 2 because we are selecting 1 dataset in the filtering panel, so we should only have 1 source data.

Reply:
Sounds good! Should we add a little comment to explain why it's 2 instead of 1, if we only select one dataset?